Willh rebase ci

willhr3 opened the pull request.

willhr3 committed:
ENT-3771 Add new CreateKeyForAccount and update tests following changes to CI flows.

willhr3 committed:
ENT-3771 Fix loan book flows to use new CI flow

roger3cev commented:
build

willhr3 commented on examples/cordapp-sweepstake/out/test/resources/participants.txt:
```diff
@@ -0,0 +1,32 @@
+Oliver
```
This is used in the int test.

roger3cev replied:
this is in the `out` directory though. I don't think we should check these files in!

willhr3 replied:
Ah yes - you're right. I removed all the `out` directories so it should be gone now.

willhr3 commented on workflows/src/main/kotlin/com/r3/corda/lib/accounts/workflows/flows/CreateKeyForAccount.kt:
```diff
@@ -0,0 +1,22 @@
+ * This flow should be used when you want to generate a new [PublicKey] for an account that owned by the node running
+ * the flow. If the account is owned by a different node, then [RequestKeyForAccountFlow] should be used instead.
+ */
+class CreateKeyForAccount(
```
Done

willhr3 committed:
ENT-3771 Switch flow into function to reduce overhead

roger-that-dev reviewed (approved):
Looks good to me - just a little bit of cleanup + kdocs + comments to go but looks good to go for the RC now. Cheers!

willhr3 committed:
ENT-3771 Updated README

willhr3 committed:
ENT-3771 Fix sweepstake flows

Commit:
Bumped dependency versions.
Fixed compile error.
Fixed broken tests.
Added some comments.

Commit:
Fixed broken request account flow bug.

Commit:
Added better kdocs and fixed some compiler warnings + code style things.

Commit:
Create account flow changed to allow creation of an account of the same name as another providing the host is another node.
Querying for account info by name now returns multiple accounts of the same name but with different hosts.
Added a test to prove the above works.

Commit:
Deleted account policy flows as not required for this version.
Removed unrequired HQL "helpers" and removed _explicit_ dependency on node - note there's still a dependency.
Re-implemented "accountKeys" using the new KMS table.
Re-implemented "accountInfo" taking a public key using the new KMS "externalIdForPublicKey" API.
Added test to check we can return all keys for an account.
Added TODOs for code where there is a dependency on "corda-node".

Commit:
Addressed https://github.com/corda/accounts/issues/31.

roger3cev commented:
This PR won't build until the required Corda SNAPSHOT is available in artifactory. Hopefully that will be on Monday morning...

Commit:
Added kdocs to new flows.

Commit:
Updated tests to take into account new accountInfo API which returns a list instead of nullable account info.
Fixed a broken TCS test which was now incorrect after the identity service changes in 4.3 - "identityService.assertOwnership" should be deprecated.

Commit:
* Removed dependency on node to store PK HASH -> EXT ID mappings. Also, we shouldn't store other nodes' keys in the KMS. Instead, there is now an entity in the accounts service which maps public keys of other nodes to external IDs.
* Added public key hash to account ID entity.
* Added new API on accounts service "accountIdForKey".

Commit:
Removed dependency on corda-node by using the new externalIds property in VaultQueryCriteria.
Rebased the repo to use VaultQueryCriteria.
Removed the old query helpers.

Commit:
* Updated TODOs.
* Added kdoc for accounts service.
* Removed create account API where the account ID is specified - we shouldn't do this as people can add accounts with the same ID on different nodes. Of course they can do this anyway if they access the hibernate entities directly but with the API available it's made easier.
* Made the query criteria helper which currently doesn't work a CordaInternal function. Can be removed when the query bug is fixed.

roger3cev commented:
build

Commit:
Build.gradle changes to get CI to work.

roger3cev commented:
build
